tools/dogfooding/dogfood.sh

```diff
@@ -0,0 +1,271 @@
+        echo_warn "Running in DRY RUN mode. Skipping 'gh pr create'."
+    else
+        gh pr create --title "$PR_TITLE" --body "$PR_DESCRIPTION" --draft --head "$DOGFOODING_BRANCH_NAME" --base "$target_branch"
+    fi
+    cd -
+}
+
+# Resolves dependencies version in `dd-sdk-ios`.
+resolve_dd_sdk_ios_package() {
+    echo_subtitle "Resolve dd-sdk-ios package in '$SDK_PACKAGE_PATH'"
+    swift package --package-path "$SDK_PACKAGE_PATH" resolve
+    echo "dd-sdk-ios dependencies:"
+    swift package --package-path "$SDK_PACKAGE_PATH" show-dependencies
+}
+
+# Reads sdk_version from current commit and stores it in DOGFOODED_SDK_VERSION variable.
+read_dogfooded_version() {
+    echo_subtitle "Read sdk_version from '$SDK_VERSION_FILE_PATH'"
+    sdk_version=$(grep 'internal let __sdkVersion' "$SDK_VERSION_FILE_PATH" | awk -F'"' '{print $2}')
+    DOGFOODED_SDK_VERSION="$sdk_version-$DOGFOODED_COMMIT_SHORT" # e.g. '2.14.1-b5215c02'
+    echo_info "▸ SDK version is '$sdk_version'"
+    echo_succ "▸ Using '$DOGFOODED_SDK_VERSION' for dogfooding"
+}
+
+# Reads the hash of previously dogfooded commit to LAST_DOGFOODED_COMMIT variable.
+read_last_dogfooded_commit() {
+    local package_resolved_path="$1"
+    echo_subtitle "Get last dogfooded commit hash in '$REPO_NAME' repo"
+    output=$(make run PARAMS="read-dogfooded-hash.py --repo-package-resolved-path '$package_resolved_path'")
+    echo "$output"
+    LAST_DOGFOODED_COMMIT=$(echo "$output" | awk -F 'DOGFOODED_HASH=' '/DOGFOODED_HASH/ {print $2}')
+}
+
+# Prints changes since LAST_DOGFOODED_COMMIT.
+print_changelog_since_last_dogfooded_commit() {
+    echo_subtitle "Generate changelog" >&2
+
+    if [ "$CI" = "true" ]; then
+        # Fetch branch history and unshallow in GitLab which only does shallow clone by default
+        echo_info "▸ Fetching git history " >&2
+        git fetch -q --unshallow >&2
+    fi
+
+    # Read git history from last dogfooded commit to current:
+    echo_info "▸ Reading commits ($LAST_DOGFOODED_COMMIT..HEAD):" >&2
+    git_log=$(git --no-pager log \
+                    --pretty=oneline "$LAST_DOGFOODED_COMMIT..HEAD" \
+                    --ancestry-path "origin/$DOGFOODED_BRANCH"
+    )
+    echo_info ">>> git log begin" >&2
+    echo "$git_log" >&2
+    echo_info "<<< git log end" >&2
+
+    # Extract only merge commits:
+    CHANGELOG=$(echo "$git_log" | grep -o 'Merge pull request #[0-9]\+' | awk -F'#' '{print "- https://github.com/DataDog/dd-sdk-ios/pull/"$2}' || true)
+    if [ -z "$CHANGELOG" ]; then
+        CHANGELOG="- Empty (no PRs merged since https://github.com/DataDog/dd-sdk-ios/commit/$LAST_DOGFOODED_COMMIT)"
+    fi
+
+    echo_info "▸ Changelog:" >&2
+    echo_info ">>> changelog begin" >&2
+    echo_succ "$CHANGELOG" >&2
+    echo_info "<<< changelog end" >&2
+
+    echo "$CHANGELOG"
+}
+
+# Updates dd-sdk-ios version in dependant project to DOGFOODED_COMMIT.
+update_dependant_package_resolved() {
```

Comment:
/improvement

given we are always going to have `develop` as target branch, we should use `xcodebuild -resolvePackageDependencies`. This is future proof solution and we don't have to maintain updating Package.resolved file logic.

This may eventually update all the dependencies but that something is expected because that's how they are being configured in the project. (minor, patch upgrade rules).

Reply:
Unless I miss something I think this could only work for Shopist which only depends on `dd-sdk-ios`. It won't work for Datadog app which has dozens of other dependencies and uses [custom build system](https://github.com/DataDog/datadog-ios/blob/develop/README.md#build). We don't want to dictate the way of managing dependencies in `datadog-ios`.

Reply:
this is actually a blessing and curse, now our SDK has dependencies on other SPM packages, we want them also get updated when `dd-sdk-ios` updates them. I think, the current solution won't work.

Also, this is going to make PR and we are going to review. So should be fine to spot things that are not looking good. If the other dozens of dependencies we have are not suppose to be updated, we must use `exact` condition to version control.

Reply:
and probably the reality is, when you do xcodebuild, it is eventually going to update the dependencies anyway. We are not gaining anything here.

Reply:
> now our SDK has dependencies on other SPM packages, we want them also get updated when dd-sdk-ios updates them. I think, the current solution won't work

We [have this covered](https://github.com/DataDog/dd-sdk-ios/blob/65854b3fe0a2aff3dec470e3f74835d1abffe2c7/tools/distribution/dogfood.py#L62-L81) for a long time and it is [migrated as part of this PR](https://github.com/DataDog/dd-sdk-ios/blob/201b40247c518297de875597415402f019c61327/tools/dogfooding/update-dependency.py#L37-L55) with no change. In terms of updating `Package.resolved` in dependent projects, this PR doesn't bring any new logic. 

> If the other dozens of dependencies we have are not suppose to be updated, we must use exact condition to version control

Again, this is asking to change process of dependency management in different repo (`datadog-ios`) and another team, which is not something we can conclude here. I get the improvement idea, but it doesn't seem doable in this PR.

Reply:
If `datadog-ios` team is not using `disableAutomaticPackageResolution` flag during xcodebuild, they are already doing what I'm proposing, but I also agree on the ownership ambiguity.